platform/src/test/kotlin/org/stellar/anchor/platform/payment/observer/stellar/PaymentObservingAccountsManagerTest.kt

```diff
@@ -0,0 +1,181 @@
+package org.stellar.anchor.platform.payment.observer.stellar
+
+import io.mockk.*
+import io.mockk.impl.annotations.MockK
+import java.time.Duration
+import java.time.Instant
+import java.time.temporal.ChronoUnit.DAYS
+import java.time.temporal.ChronoUnit.HOURS
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.stellar.anchor.platform.data.PaymentObservingAccount
+
+class PaymentObservingAccountsManagerTest {
+  @MockK private lateinit var paymentObservingAccountStore: PaymentObservingAccountStore
+
+  @BeforeEach
+  fun setUp() {
+    MockKAnnotations.init(this, relaxed = true)
+    every { paymentObservingAccountStore.list() } returns
+      listOf(
+        PaymentObservingAccount(
+          "GCK5ECMM67ZN7RWGUSDKQAW6CAF6Q5WYK2VQJ276SJMINU6WVCIQW6BL",
+          Instant.now().minus(0, HOURS)
+        ),
+        PaymentObservingAccount(
+          "GCIWQDKACLW26UJXY5CTLULVYUOYROZPAPDDYEQKNGIERVOAXSPLABMB",
+          Instant.now().minus(2, HOURS)
+        ),
+        PaymentObservingAccount(
+          "GAPBFA5ZYG5VVKN7WPMH6K5CBXGU2AM5ED7S54VX27J7S222NKMTWKR6",
+          Instant.now().minus(12, HOURS)
+        )
+      )
+  }
+
+  @AfterEach
+  fun teardown() {
+    clearAllMocks()
+    unmockkAll()
+  }
+
+  @Test
+  fun test_addAndRemove_success() {
+    val obs = PaymentObservingAccountsManager(paymentObservingAccountStore)
+    obs.initialize()
+
+    obs.upsert("GCIWQDKACLW26UJXY5CTLULVYUOYROZPAPDDYEQKNGIERVOAXSPLABMB", true)
+    obs.upsert("GCK5ECMM67ZN7RWGUSDKQAW6CAF6Q5WYK2VQJ276SJMINU6WVCIQW6BL", true)
+    obs.upsert("GAPBFA5ZYG5VVKN7WPMH6K5CBXGU2AM5ED7S54VX27J7S222NKMTWKR6", true)
+
+    assertEquals(3, obs.accounts.size)
+
+    obs.remove(null)
+    assertEquals(3, obs.accounts.size)
+
+    obs.remove("GCIWQDKACLW26UJXY5CTLULVYUOYROZPAPDDYEQKNGIERVOAXSPLABMB")
+    obs.remove("GCIWQDKACLW26UJXY5CTLULVYUOYROZPAPDDYEQKNGIERVOAXSPLABMB")
```

Comment:
Did you intend to remove the same account twice here?

Reply:
Yes. I want to make sure the second time does not cause failure.

Reply:
Cool, thanks for clarifying that 😄